internal/ingress/controller/endpoints.go

```diff
@@ -113,6 +118,8 @@ func getEndpoints(s *corev1.Service, port *corev1.ServicePort, proto corev1.Prot
 				upsServers = append(upsServers, ups)
 				processedUpstreamServers[ep] = struct{}{}
 			}
+			// only one port could be matched, no need to iterate remaining ports
+			break
```

Comment:
Thanks for your contributions. The first thing to confirm is whether there will be side effects here

Reply:
All e2e tests passed. What can i do next?

Reply:
I'm not sure if there could be a endpoint subsets with different port number and same port name. I think it will not appear because k8s service validation will reject duplicate port name values.

Howerver, endpoint validation doesn't validate this rule. 

We can remove `break` here. It will not affect result but cost more compute resources.

What do you think?

Reply:
```yaml
// EndpointPort is a tuple that describes a single port.
// +structType=atomic
type EndpointPort struct {
    // The name of this port.  This must match the 'name' field in the
    // corresponding ServicePort.
    // Must be a DNS_LABEL.
    // Optional only if one port is defined.
    // +optional
    Name string `json:"name,omitempty" protobuf:"bytes,1,opt,name=name"`

    // The port number of the endpoint.
    Port int32 `json:"port" protobuf:"varint,2,opt,name=port"`

    ...
}

// ServicePort contains information on service's port.
type ServicePort struct {
    // The name of this port within the service. This must be a DNS_LABEL.
    // All ports within a ServiceSpec must have unique names. When considering
    // the endpoints for a Service, this must match the 'name' field in the
    // EndpointPort.
    // Optional if only one ServicePort is defined on this service.
    // +optional
    Name string `json:"name,omitempty" protobuf:"bytes,1,opt,name=name"`

    // The IP protocol for this port. Supports "TCP", "UDP", and "SCTP".
    // Default is TCP.
    // +default="TCP"
    // +optional
    Protocol Protocol `json:"protocol,omitempty" protobuf:"bytes,2,opt,name=protocol,casttype=Protocol"`

    ...
}
```

Reply:
I pushed a new commit to remove break here to avoid unknown problem.

Reply:
@tao12345666333 Do you have time to review this PR?

Reply:
sorry for delay. I will review it today.

/assign